### PR TITLE
chore(ci): Have Vector `master` use `main` of VRL

### PR DESCRIPTION
**Pull request by jszwedko**
So that nightly releases can you unreleased VRL changes for testing. This was requested by a user. I can see the usefulness to be able to validate VRL changes without having to build Vector.

As part of the release, we'll switch it to the latest released version (which should be the same SHA).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -149,7 +149,7 @@ toml = { version = "0.8.19", default-features = false, features = ["display", "p
 tonic = { version = "0.11", default-features = false, features = ["transport", "codegen", "prost", "tls", "tls-roots", "gzip"] }
 tonic-build = { version = "0.11", default-features = false, features = ["transport", "prost"] }
 uuid = { version = "1.10.0", features = ["v4", "v7", "serde"] }
-vrl = { version = "0.19.0", features = ["arbitrary", "cli", "test", "test_framework"] }
+vrl = { git = "https://github.com/vectordotdev/vrl", branch = "main", features = ["arbitrary", "cli", "test", "test_framework"] }
```
Comment (pront) [inline Cargo.toml:152]: Do we have any mitigation for the following scenario?

1. Vector builds successfully with the latest VRL main
2. We merge a new commit that changes an interface in VRL
3. Vector builds will fail until we update the code

Comment (jszwedko) [reply Cargo.toml:152]: The version will still be locked so an update to VRL won't break the Vector build on `master`. Instead, a PR opened by dependabot that bumps VRL to latest will fail at which point we can intervene to make the necessary Vector updates in the same PR.

**Review by pront — APPROVED** [review]
Thanks!

**Comment by jszwedko** [comment]
FYI @johntodd I finally was able to get this through :) Nightly Vector releases should now be using latest VRL (generally, dependabot will be keeping it up-to-date so it may lag by a commit or two at times).

**Comment by jszwedko** [comment]
FYI @johnhtodd I finally was able to follow up on this :) Nightly Vector releases should now be using latest VRL, generally. Dependabot will be keeping it up-to-date so it may lag by a commit or two at times.

**Comment by johnhtodd** [comment]
@jszwedko  Great!  Thanks. That will speed testing significantly.

**Pull request merged**